lib/service/storage/elasticsearch.js

```diff
@@ -2620,15 +2677,124 @@ class ElasticSearch extends Service {
+   */
+  async _getESIndexFromAlias (index, collection) {
+    const alias = `${ALIAS_PREFIX}${this._indexPrefix}${index}${NAME_SEPARATOR}${collection}`;
+    const { body } = await this._client.cat.aliases({ format: 'json', name: alias });
+
+    if (body.length < 1) {
+      throw kerror.get('unknown_index_collection');
+    }
+    else if (body.length > 1) {
+      throw kerror.get('multiple_index_alias', 'alias', 'index');
+    }
+
+    return body[0].index;
+  }
+
+  /**
+   * Gets the alias name from a concrete esIndex
+   *
+   * @param {String} esIndex
+   *
+   * @returns {String} alias name (eg: '@&nepali.liia')
+   * @throws If there is not exactly one alias related to this esIndex
+   */
+  async _getAliasFromESIndex (esIndex) {
+    const { body } = await this._client.indices.getAlias({ index: esIndex});
+    const aliases = Object.keys(body[esIndex].aliases);
+
+    if (aliases.length < 1) {
+      throw kerror.get('unknown_index_collection');
+    }
+    else if (aliases.length > 1) {
+      throw kerror.get('multiple_index_alias', 'index', 'alias');
+    }
+
+    return aliases[0];
+  }
+
+  /**
+   * Gets the concrete esIndex name from an alias (index + collection)
+   *
+   * @param {String} esIndex
+   *
+   * @returns {boolean} `true` if the raw ES index name is available, `false` otherwise
+   */
+  async _isESIndexAvailable (esIndex) {
+    return ! await this._client.indices.exists({ index: esIndex }).body;
+  }
+
+  /**
+   * Gets the concrete esIndex name from an alias (index + collection)
+   *
+   * @param {String} index
+   * @param {String} collection Default to `HIDDEN_COLLECTION`
+   *
+   * @returns {String} Available esIndex name (eg: '&nepali.liia')
+   */
+  async _getSafeESIndex (index, collection = HIDDEN_COLLECTION) {
+    let esIndex = this._getESIndex(index, collection).substr(INDEX_PREFIX_POSITON_IN_ALIAS);
+    let available = await this._isESIndexAvailable(esIndex);
+
+    while (! available) {
+      esIndex = generateRandomCustom(
+        (adjective, name) => `${this._indexPrefix}${adjective}${NAME_SEPARATOR}${name}`);
+
+      available = Buffer.from(esIndex).length < 255 && await this._isESIndexAvailable(esIndex);
+    }
+
+    return esIndex;
+  }
+
+  /**
+   * Check for each ES index whether it has an alias or not.
+   * When the latter is missing, create one based on the actual index name.
+   *
+   * This check avoids a breaking change for those who were using Kuzzle before
+   * alias attribution for each ES index turned into a standard ('auto-version').
+   */
+  async _ensureAliasConsistency () {
+    try {
+      const { body } = await this._client.cat.indices({ format: 'json' });
+      const esIndexes = body.map(({ index }) => index);
+      const aliases = await this.listAliases();
+
+      const indexesWithoutAlias = esIndexes.filter((index) =>
+        index[INDEX_PREFIX_POSITON_IN_INDEX] === this._indexPrefix
+        && ! aliases.some((alias) => alias.name === index));
```

Comment:
Nit: coding standard
```suggestion
      const indexesWithoutAlias = esIndexes.filter(index =>
              index[INDEX_PREFIX_POSITON_IN_INDEX] === this._indexPrefix
        && ! aliases.some(alias => alias.name === index));
```